### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: scripts/restore [hunk 1]
```diff
@@ -46,13 +46,6 @@ ynh_restore_file --origin_path="/etc/logrotate.d/$app"
 
 yunohost service add $app --description="VS Code Server" --log="/var/log/$app/$app.log"
```
Comment (Tagadda) [inline scripts/restore:48]:
```suggestion
chown -R $admin:$admin "$data_dir"
```

File: manifest.toml [hunk 2]
```diff
@@ -60,3 +59,4 @@ ram.runtime = "100M"
 
     [resources.permissions]
     main.url = "/"
+    main.allowed = "admins"
```
Comment (Tagadda) [inline manifest.toml:62]:
```suggestion
```

File: scripts/install [hunk 3]
```diff
@@ -60,7 +60,6 @@ chown root:$admin "$install_dir/code-server.env"
 #=================================================
 # SET PERMISSION FOR THE USER
 #=================================================
```
Comment (Tagadda) [inline scripts/install:62]:
```suggestion
#=================================================
ynh_permission_update --permission="main" --add="$admin"
```

File: scripts/restore [hunk 4]
```diff
@@ -45,14 +45,7 @@ mkdir -p "/var/log/$app"
 ynh_restore_file --origin_path="/etc/logrotate.d/$app"
 
 yunohost service add $app --description="VS Code Server" --log="/var/log/$app/$app.log"
```
Comment (Tagadda) [inline scripts/restore:47]:
```suggestion
yunohost service add $app --description="VS Code Server" --log="/var/log/$app/$app.log"

ynh_permission_update --permission="main" --add="$admin"
```

File: scripts/upgrade [hunk 5]
```diff
@@ -111,30 +103,20 @@ chown $admin:$admin "$install_dir/config.yaml" "$install_dir/code-server.env"
...
-#=================================================
-# INTEGRATE SERVICE IN YUNOHOST
-#=================================================
-ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
-
 yunohost service add $app --description="VS Code Server" --log="/var/log/$app/$app.log"
 
 #=================================================
 # SET PERMISSION FOR THE USER
 #=================================================
```
Comment (Tagadda) [inline scripts/upgrade:119]:
```suggestion
#=================================================
ynh_permission_update --permission="main" --add="$admin"
```

**Review by Tagadda — CHANGES_REQUESTED** [review]

**Comment by Tagadda** [comment]
!testme

**Comment by ericgaspar** [comment]
You basically reverted all changes of this PR 😅

**Comment by Tagadda** [comment]
Yes, here code-server is run as the user selected during the install.
We don't want any other user accessible the app.

**Comment by ericgaspar** [comment]
Yes, but surly there is a way to achieve that without using `ynh_permission_update`

**Comment by Tagadda** [comment]
With manifest v2, not that I know of...
The question `init_main_permissions` only allows for groups to be selected.

**Comment by ericgaspar** [comment]
what about `main.allowed = "admins"`

**Comment by Tagadda** [comment]
It's not the same at all.
`main.allowed = "admins"` will grant access to the `admins` group
`ynh_permission_update --permission="main" --add="$admin"` will grant access to the user selected during install

**Comment by Tagadda** [comment]
!testme

**Review by Tagadda — APPROVED** [review]

**Pull request merged**